### PR TITLE
added new spin transition example

### PR DESCRIPTION
**Pull request by bobmyhill**
This PR adds a new spin transition example for finite temperatures.

### CODE REVIEW

File: examples/example_spintransition_thermal.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+example_spintransition_thermal
+------------------------------
+
+This example illustrates how to create a non-ideal solid solution model
+for (Mg,FeHS,FeLS)O that has a gradual spin transition at finite temperature.
+First, we define the MgO endmember and two endmembers for the
+low and high spin states of FeO. Then we create a regular/symmetric
+solid solution that incorporates all three endmembers.
+The solid solution is contains a method called set_equilibrium_composition,
```
Comment (tjhei) [inline examples/example_spintransition_thermal.py:16]:
```suggestion
The solid solution contains a method called set_equilibrium_composition,
```

File: examples/example_spintransition_thermal.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+which calculates the equilibrium proportions of the low and high spin phases
+at the desired bulk composition, pressure and temperature.
+
+In this example, we are neglecting the elastic component of mixing. We are
+also implicitly applying the Bragg-Williams approximation (i.e. we assume
```
Comment (tjhei) [inline examples/example_spintransition_thermal.py:21]:
```suggestion
also implicitly applying the Bragg-Williams approximation (i.e., we assume
```

File: examples/example_spintransition_thermal.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            # finally, set the equilibrium composition that we have
+            # just calculated
+            self.set_composition([1. - molar_fraction_FeO,
+                                  molar_fraction_FeO*(1. - p_LS),
+                                  molar_fraction_FeO*p_LS])
+
+    # In this line, we create our solid solution object
+    fper = ferropericlase()
+
+    """
```
Comment (tjhei) [inline examples/example_spintransition_thermal.py:145]: should this be a comment block starting with ``#``?

**Review by tjhei — APPROVED** [review]

**Comment by sannecottaar** [comment]
Nice example, Bob.

**Pull request merged**